lightningd/log.c

```diff
@@ -564,40 +564,66 @@ void logv(struct logger *log, enum log_level level,
 	  const char *fmt, va_list ap)
 {
 	int save_errno = errno;
-	struct log_entry *l = new_log_entry(log, level, node_id);
 	char *log_msg = NULL;
 
 	/* This is WARN_UNUSED_RESULT, because everyone should somehow deal
 	 * with OOM, even though nobody does. */
 	if (vasprintf(&log_msg, fmt, ap) == -1)
 		abort();
 
-	char **lines = tal_strsplit(
-	    log, json_escape_unescape(log, (struct json_escape *)log_msg), "\n",
-	    STR_NO_EMPTY);
+	const char *unescaped_log =
+	    json_escape_unescape(NULL, (struct json_escape *)log_msg);
```

Comment:
This is the cause of your leak, BTW.  Every tal allocation is off some other tal allocation (or NULL).  There's a special throwaway one called `tmpctx` which automagically gets cleaned in the event loop.  Since you used NULL here, and don't free it, it gets caught as a leak by our leak detection code (and it is!).